client/gutenberg/extensions/simple-payments/product-placeholder.jsx

```diff
@@ -19,6 +28,15 @@ export default ( { title = '', content = '', formattedPrice = '', multiple = fal
 	return (
 		<div className="jetpack-simple-payments-wrapper">
 			<div className="jetpack-simple-payments-product">
+				{ featuredMedia && (
+					<div class="jetpack-simple-payments-product-image">
+						<div class="jetpack-simple-payments-image">
+							<RawHTML>
+								{ featuredMedia.description.rendered.replace( 'class="attachment"', '' ) }
+							</RawHTML>
```

Comment:
Using `RawHTML` just for the proof of concept, we might want to construct image tag here instead.